pkg/dyninst/module/persistent_upload_cache.go

```diff
@@ -0,0 +1,282 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+package module
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+)
+
+// entryType represents the type of upload entry.
+type entryType int
+
+const (
+	// entryTypeAttempt represents an upload attempt entry. An attempt is
+	// inserted in the cache before an upload is started; the entry is then
+	// converted to a completed one if the upload is successful.
+	entryTypeAttempt entryType = iota // attempt
```

Comment:
nit: start the meaningful values at 1 -- it's too easy in go to have a zero-valued field you didn't mean to initialize to a meaningful value

Reply:
done